emcc.py

```diff
@@ -2975,6 +2970,67 @@ def generate_html(target, options, js_target, target_basename,
     f.write(asbytes(html_contents))
 
 
+def minify_html(filename, options):
+  opts = []
+  # -g1 and greater retain whitespace and comments in source
+  if options.debug_level == 0:
+    opts += ['--collapse-whitespace',
+             '--collapse-inline-tag-whitespace',
+             '--remove-comments',
+             '--remove-tag-whitespace',
+             '--sort-attributes',
+             '--sort-class-name']
+  # -g2 and greater do not minify HTML at all
+  if options.debug_level <= 1:
+    opts += ['--decode-entities',
+             '--collapse-boolean-attributes',
+             '--remove-attribute-quotes',
+             '--remove-redundant-attributes',
+             '--remove-script-type-attributes',
+             '--remove-style-link-type-attributes',
+             '--use-short-doctype',
+             '--minify-css', 'true',
+             '--minify-js', 'true']
+
+  # html-minifier also has the following options, but they look unsafe for use:
+  # '--remove-optional-tags': removes e.g. <head></head> and <body></body> tags from the page.
+  #                           (Breaks at least browser.test_sdl2glshader)
+  # '--remove-empty-attributes': removes all attributes with whitespace-only values.
+  #                              (Breaks at least browser.test_asmfs_hello_file)
+  # '--remove-empty-elements': removes all elements with empty contents.
+  #                            (Breaks at least browser.test_asm_swapping)
+
+  if options.debug_level >= 2:
+    return
+
+  logger.debug('minifying HTML file ' + filename)
+  size_before = os.path.getsize(filename)
+  cmd = shared.NODE_JS + [shared.path_from_root('third_party', 'html-minifier', 'cli.js'), filename, '-o', filename] + opts
+  logger.debug(str(cmd))
```

Comment:
this is not needed, `run_process` will log if debugging.

Reply:
check